core/tracker/inclusion.go

```diff
@@ -632,8 +647,37 @@ func (a *InclusionChecker) Run(ctx context.Context) {
 			if checkedSlot == slot {
 				continue
 			}
+			if checkedEpoch != slot/slotsPerEpoch {
+				epoch := slot / slotsPerEpoch
+				indices := []eth2p0.ValidatorIndex{}
+				a.core.mu.Lock()
+				subs := a.core.submissions
```

Comment:
This is wrong:
> You are still accessing the original map, not a real copy of its contents. If another goroutine writes to the map at the same time, this may cause: a) race conditions, b) runtime panic: concurrent map read and map write, if writes occur concurrently with reads

Maybe here we start using brand new `sync.Map` from golang 1.24? and avoid mutex entirely?

Reply:
Ah, you are right! I've cloned it.

I'm a bit reluctant on refactoring for this bugfix... I know also the `sync.Map` is a bit slower. We can probably think about it later?